resourcet/Data/Acquire/Internal.hs

```diff
@@ -25,8 +26,22 @@ import qualified Control.Monad.Catch as C ()
 -- @since 1.1.2
 data ReleaseType = ReleaseEarly
                  | ReleaseNormal
-                 | ReleaseException
-    deriving (Show, Read, Eq, Ord, Enum, Bounded, Typeable)
+                 | ReleaseExceptionWith E.SomeException
+    deriving (Show)
+
+instance Eq ReleaseType where
+    ReleaseEarly == ReleaseEarly = True
+    ReleaseNormal == ReleaseNormal = True
+    ReleaseExceptionWith (E.SomeException e0) == ReleaseExceptionWith (E.SomeException e1) =
+        case typeOf e0 == typeOf e1 of
+            True ->
+                show e0 == show e1
```

Comment:
Is this expected to be right? :grimacing:

Reply:
I've used it a bunch and it uhhh does mostly work

Reply:
Until future changes to GHC render the backtrace in show?

Reply:
No, those would only be able to work on the `show :: SomeException -> String` type, since `SomeException` carries the implicit parameters for the stack trace. This uses the `Show a` that is implied by the `Exception a` dictionary that is packed in the GADT.